libcodechecker/analyze/analyzers/analyzer_clang_tidy.py

```diff
@@ -116,3 +117,41 @@ def construct_analyzer_cmd(self, res_handler):
         except Exception as ex:
             LOG.error(ex)
             return []
+
+    @classmethod
+    def resolve_missing_binary(cls, configured_binary, env):
```

Comment:
No, there isn't. We attempted the `distutils.spawn.find_executable` but it can only match exact binary names.

Also right now it seems that the code is a duplicate, it _is_, but I would vouch **against** abstracting it out: it would create an extra layer of abstraction which doesn't really fit into the model we are using.

`BaseAnalyzer`
`RegexMatchableInstallNamedAnalyzer : public BaseAnalyzer`
`ClangSA : public RegexMatchableInstallNamedAnalyzer`

Any time there comes a new analyzer (we are still advertising CodeChecker to be compatible with many analyzers even though we only officially support clangsa and -tidy) which cannot be matched on any of the regex patters and we would break this chain. In my view, a new layer of abstraction here would only cause more clutter than how much imminent duplication it could solve.